bgp: remove metallb-bgp documentation (#36306)
cilium-team/cilium: Documentation/operations/upgrade.rst

* The previously deprecated built-in WireGuard userspace-mode fallback (Helm ``wireguard.userspaceFallback``)
  has been removed. Users of WireGuard transparent encryption are required to use a Linux kernel with
  WireGuard support.

> * The previously deprecated ``metallb-bgp`` flags ``bgp-config-path``, ``bgp-announce-lb-ip``
>   and ``bgp-announce-pod-cidr`` has been removed. Users are required to use Cilium BGP
>   control plane for BGP advertisements.

Reviewer: ```suggestion
* The deprecated ``metallb-bgp`` flags ``bgp-config-path``, ``bgp-announce-lb-ip``
  and ``bgp-announce-pod-cidr`` have been removed. Users are now required to use Cilium BGP
  control plane for BGP advertisements.
```

  to validate certificates with expirations longer than 825 days.
* The Helm option ``hubble.relay.dialTimeout`` has been deprecated (now a no-op)
  and will be removed in Cilium 1.18.
* The ``metallb-bgp`` integration Helm options ``bgp.enabled``, ``bgp.announce.podCIDR``, and

>   ``bgp.announce.loadbalancerIP`` has been removed. Users are required to use Cilium BGP

Reviewer: ```suggestion
  ``bgp.announce.loadbalancerIP`` have been removed. Users are now required to use Cilium BGP
```
